Source/WebCore/platform/graphics/ComplexTextController.cpp

```diff
@@ -380,6 +389,12 @@ void ComplexTextController::collectComplexTextRuns()
 
         nextFont = m_font.fontForCombiningCharacterSequence(baseOfString.subspan(previousIndex, currentIndex - previousIndex));
 
+        if (shouldProcessTextSpacingTrim && nextFont && !nextFont->isSystemFontFallbackPlaceholder()) {
+            TextSpacing::CharactersData charactersData = { .currentCharacter = baseCharacter, .currentCharacterClass = TextSpacing::characterClass(baseCharacter) };
+            halfWidthFont = TextSpacing::getHalfWidthFontIfNeeded(*nextFont, m_font.textSpacingTrim(), charactersData);
+            nextFont = halfWidthFont ? halfWidthFont : nextFont;
+        }
```

Comment:
Seems like we have some repition here. Is there an opportunity to refactor here?

Reply:
(Just a minor observation and no concrete suggestions)